common/src/lib/java/dev/engine_room/flywheel/lib/model/part/VertexWriter.java

```diff
@@ -23,62 +23,60 @@ public VertexWriter() {
 	}
 
 	@Override
-	public VertexConsumer vertex(double x, double y, double z) {
+	public VertexConsumer addVertex(float x, float y, float z) {
 		if (!filledPosition) {
 			long ptr = vertexPtr();
-			MemoryUtil.memPutFloat(ptr, (float) x);
-			MemoryUtil.memPutFloat(ptr + 4, (float) y);
-			MemoryUtil.memPutFloat(ptr + 8, (float) z);
+			MemoryUtil.memPutFloat(ptr, x);
+			MemoryUtil.memPutFloat(ptr + 4, y);
+			MemoryUtil.memPutFloat(ptr + 8, z);
 			filledPosition = true;
 		}
-		return this;
+		return endVertexIfNeeded();
 	}
 
 	@Override
-	public VertexConsumer color(int red, int green, int blue, int alpha) {
-		// ignore color
-		return this;
+	public VertexConsumer setColor(int red, int green, int blue, int alpha) {
+		return endVertexIfNeeded();
 	}
 
 	@Override
-	public VertexConsumer uv(float u, float v) {
+	public VertexConsumer setUv(float u, float v) {
 		if (!filledTexture) {
 			long ptr = vertexPtr();
 			MemoryUtil.memPutFloat(ptr + 12, u);
 			MemoryUtil.memPutFloat(ptr + 16, v);
 			filledTexture = true;
 		}
-		return this;
+		return endVertexIfNeeded();
 	}
 
 	@Override
-	public VertexConsumer overlayCoords(int u, int v) {
+	public VertexConsumer setUv1(int u, int v) {
 		// ignore overlay
-		return this;
+		return endVertexIfNeeded();
 	}
 
 	@Override
-	public VertexConsumer uv2(int u, int v) {
+	public VertexConsumer setUv2(int u, int v) {
 		// ignore light
-		return this;
+		return endVertexIfNeeded();
 	}
 
 	@Override
-	public VertexConsumer normal(float x, float y, float z) {
+	public VertexConsumer setNormal(float x, float y, float z) {
 		if (!filledNormal) {
 			long ptr = vertexPtr();
 			MemoryUtil.memPutByte(ptr + 20, DataPacker.packNormI8(x));
 			MemoryUtil.memPutByte(ptr + 21, DataPacker.packNormI8(y));
 			MemoryUtil.memPutByte(ptr + 22, DataPacker.packNormI8(z));
 			filledNormal = true;
 		}
-		return this;
+		return endVertexIfNeeded();
 	}
 
-	@Override
-	public void endVertex() {
+	public VertexConsumer endVertexIfNeeded() {
 		if (!filledPosition || !filledTexture || !filledNormal) {
-			throw new IllegalStateException("Not filled all elements of the vertex");
+			return this;
```

Comment:
Will there be new subtle bugs in dev instead of exceptions because we never throw? Does this mimic some changed vanilla behavior? If so please add a comment explaining it

Reply:
Vanilla no longer throws a exception